pykafka/simpleconsumer.py

```diff
@@ -828,7 +829,18 @@ def enqueue_messages(self, messages):
         :type messages: Iterable of :class:`pykafka.common.Message`
         """
         for message in messages:
-            if message.offset != self.next_offset:
+            # HAck to deal with compacted topic not returning earliest offset
+            # correctly
+            if not self._consumed_first_msg and message.offset != self.next_offset:
+                log.info("Detected compacted topic. "
+                          "Setting earliest offest to (%s) "
+                          "not equal to next_offset (%s)",
+                          message.offset, self.next_offset)
+                self.next_offset = message.offset
+                self._consumed_first_msg = True
```

Comment:
Thanks, I agree. @yungchin and @rduplain should chime in, but I think that a keyword argument on `SimpleConsumer` indicating whether the topic is compacted and being used to switch between `!=` and `<` is a fine stopgap to merge.